### ISSUE TITLE
Plugin: fastify-schedule

### ISSUE THREAD

**Issue by cemremengu** [fastify/fastify#1107]
Some of the frameworks I came across has a schedule plugin like [here](https://github.com/miaowing/nest-schedule) and [here](https://github.com/eggjs/egg-schedule). They usually benefit from [node-schedule](https://www.npmjs.com/package/node-schedule) or wrap their own scheduler around [cron-parser](https://github.com/harrisiirak/cron-parser).

I am thinking of attempting a plugin like this. I will probably use `node-schedule` as the underlying scheduler. Any thoughts?

**Comment by mcollina** [comment]
Go for it!

**Comment by jsumners** [comment]
I think https://twitter.com/jsumners79/status/775391136802336768 applies.

**Comment by jsumners** [comment]
I said the point of the post applies: the cron syntax for specifying scheduled jobs is archaic. No one can remember that garbage 5 minutes after they read the man page; and if someone says otherwise they are lying.

There is absolutely *no reason* a modern scheduling system should need to support that syntax. Especially when you have easy access to objects like:

```js
{
  day: 'all',
  hour: '12:00'
}
```

**Comment by cemremengu** [comment]
Yes, you are definitely right about that. To be honest, I sometimes don't understand cron schedules even while looking at the man page to decipher it 😄 However, it's the most expressive way to represent schedules so far. For example how would you represent `last 3 days of every month` or `every week except second` in that object notation you mentioned (yes we have clients that request schedules like these unfortunately). Even if you could,  I think you would end up something similar to cron's ugly syntax.

The only other way I can think is writing schedules in plain english sentences. What do you think about that approach?

**Comment by jsumners** [comment]
All I'm saying is to think about it.

**Comment by kibertoad** [comment]
Since there still doesn't seem to be anything for fastify to cover this, I'll pick this one up.

**Comment by kibertoad** [comment]
@jsumners node-schedule seems pretty dead. I've pinged their maintainer offering help, but if they don't react, there doesn't seem to be any decent libraries offering hybrid cron and non-cron scheduling, so I'm currently considering https://github.com/jakubroztocil/rrule as the main scheduling engine. Do you think that not having cron support will be a dealbreaker for many users? Should we use two libs, one for cron, one for rrule?

**Comment by jsumners** [comment]
Wow! I did not know that package exists. Finally, someone other than me that got fed up with the antiquated and mysterious cron syntax and actually did something about it. If someone is dead set on using cron syntax in their application, let them write a utility to convert that syntax in to the object to pass in.
